src/slider.tsx

```diff
@@ -1,164 +1,428 @@
+        const [isThumbHovered, setIsThumbHovered] = React.useState(false);
+        const [isFocused, setIsFocused] = React.useState(false);
+        const {isIos} = useTheme();
+
+        const isPointerOverElement = (element: HTMLElement | null, x: number, y: number) => {
+            const box = element?.getBoundingClientRect();
+            return box !== undefined && box.left <= x && x <= box.right && box.top <= y && y <= box.bottom;
+        };
+
+        const isTabKeyDownRef = React.useRef(false);
 
-            setValueRanger(valueIndex);
-        } else {
-            setMinSlider(min);
-            setMaxSlider(max);
-            let ranger = min;
-            if (value !== undefined) {
-                if (min > value || value > max) {
-                    ranger = max < value ? max : min;
+        React.useEffect(() => {
+            const handleKeyDown = (e: KeyboardEvent) => {
+                switch (e.keyCode) {
+                    case TAB:
+                        isTabKeyDownRef.current = true;
+                        break;
+                    case ESC:
+                        if (isFocused) {
+                            setIsFocused(false);
+                            thumbRef.current?.blur();
+                        }
+                        break;
+                    default:
+                    // do nothing
                 }
+            };
+
+            const handleKeyUp = () => (isTabKeyDownRef.current = false);
+
+            document.addEventListener('keydown', handleKeyDown, false);
+            document.addEventListener('keyup', handleKeyUp, false);
+            return () => {
+                document.removeEventListener('keydown', handleKeyDown, false);
+                document.removeEventListener('keyup', handleKeyUp, false);
+            };
+        }, [isFocused]);
+
+        const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+            switch (e.keyCode) {
+                case RIGHT:
+                case UP:
+                    cancelEvent(e);
+                    handleChange(finalValue + step, false);
+                    break;
+                case LEFT:
+                case DOWN:
+                    cancelEvent(e);
+                    handleChange(finalValue - step, false);
+                    break;
+                case HOME:
+                    cancelEvent(e);
+                    handleChange(min, false);
+                    break;
+                case END:
+                    cancelEvent(e);
+                    handleChange(max, false);
+                    break;
+
+                default:
+                // do nothing
             }
+        };
 
-            setValueRanger(ranger);
-            setStep(steps);
-        }
-    }, [steps, max, min, getApproximation, value, step, getOrder]);
+        const isTouchableDevice = isClientSide() ? window.matchMedia('(pointer: coarse)').matches : false;
```

Comment:
perhaps we could move this `isTouchableDevice` to `utils/environment`. The same is implemented `Tooltip`

Reply:
done